tutorials/dataframe/df103_NanoAODHiggsAnalysis.py

```diff
@@ -0,0 +1,574 @@
+## \file
+## \ingroup tutorial_dataframe
+## \notebook -draw
+## This tutorial is a simplified but yet complex example of an analysis
+## reconstructing the Higgs boson decaying to two Z bosons from events with
+## four leptons. The data and simulated events are taken from CERN OpenData
+## representing a subset of the data recorded in 2012 with the CMS detector at
+## the LHC. The tutorials follows the Higgs to four leptons analysis published
+## on CERN Open Data portal
+## ([10.7483/OPENDATA.CMS.JKB8.RR42](http://opendata.cern.ch/record/5500)).
+## The resulting plots show the invariant mass of the selected four lepton
+## systems in different decay modes (four muons, four electrons and two of each
+## kind) and in a combined plot indicating the decay of the Higgs boson with a
+## mass of about 125 GeV.
+##
+## The following steps are performed for each sample with data and simulated
+## events in order to reconstruct the Higgs boson from the selected muons and
+## electrons:
+## 1. Select interesting events with multiple cuts on event properties, e.g.,
+##    number of leptons, kinematics of the leptons and quality of the tracks.
+## 2. Reconstruct two Z bosons of which only one on the mass shell from the
+##    selected events and apply additional cuts on the reconstructed objects.
+## 3. Reconstruct the Higgs boson from the remaining Z boson candidates and
+##    calculate its invariant mass.
+##
+## Another aim of this version of the tutorial is to show a way to blend C++
+## and Python code. All the functions that make computations on data to define
+## new columns or filter existing ones in a precise way, better suited to be
+## written in C++, have been moved to a header that is then declared to the
+## ROOT C++ interpreter. The functions that instead create nodes of the
+## computational graph (e.g. Filter, Define) remain inside the main Python
+## script.
+##
+## \macro_image
+## \macro_code
+## \macro_output
+##
+## \date July 2019
+## \author Stefan Wunsch (KIT, CERN), Vincenzo Eduardo Padulano (UniMiB, CERN)
+
+import ROOT
+import os
+
+# Enable multi-threading
+ROOT.ROOT.EnableImplicitMT()
+
+# Include necessary header
+tutorial_dir = str(ROOT.gROOT.GetTutorialDir())
+dataframe_dir = "dataframe"
+header_file = "df103_NanoAODHiggsAnalysis_python.h"
+higgs_header_path = os.path.join(
+    os.sep, tutorial_dir + os.sep,
```

Comment:
Why do you need the `os.sep`? It should work as well without :thinking:

Reply:
`os.path.join(str(ROOT.gROOT.GetTutorialDir()), "dataframe", "df103_NanoAODHiggsAnalysis_python.h")` works for me!

Reply:
I found this suggested on stack overflow for windows-friendly usage of `os.path` ([post](https://stackoverflow.com/questions/2422798/python-os-path-join-on-windows)). I tried it on Linux and in fact there is no difference between using `os.sep` or not. Unfortunately I have no windows machine to test if this makes any difference :( Have you checked it on windows by chance?

Reply:
Ah ok, I see. Well, awkward behaviour of `os.path` since this should be their job!